benchexec/tools/sv_benchmarks_util.py

```diff
@@ -40,3 +42,114 @@ def get_data_model_from_task(task, param_dict):
                     f"Unsupported data_model '{data_model}' defined for task '{task}'"
                 )
     return None
+
+
+def _partition_input_files(task):
+    """
+    This function partitions the input files into witness and other files.
+    The distinction is based on the file name of the witness file, which
+    is identified by the option "witness" in the task options.
+
+    @param task: An instance of a task
+    @return: Tuple of witness files and other files
+    """
+    input_files = task.input_files_or_identifier
+    witness_files = []
+    other_files = []
+    for file in input_files:
+        if Path(file).name == task.options.get("witness"):
+            witness_files.append(file)
+        else:
+            other_files.append(file)
+    return witness_files, other_files
+
+
+def get_witness_input_files(task):
```

Comment:
The goal of this function is to have an abstraction layer between `_partition_input_files` and `get_unique_witness`. If you prefer I can remove it since it is only called in `get_unique_witness`.

Reply:
It should at least not clutter the public interface. But even internally I do not see that it is worth it.

Reply:
Okay, remove it :ok_hand: